src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java

```diff
@@ -106,8 +106,8 @@ public void init(ClusterAwareConnectionProxy proxy, PropertySet propertySet) {
 
   //TODO: Should method target to be passed along with method name? Like:
   // Object execute(Object methodTarget, String methodName, Callable executeSqlFunc)
-  public Object execute(String methodName, Callable executeSqlFunc) throws Exception {
-    return this.headPlugin.execute(methodName, executeSqlFunc);
+  public Object execute(Class methodInvokeOn, String methodName, Callable executeSqlFunc) throws Exception {
```

Comment:
We are calling `methodInvokeOn.getName()` a lot, do we need to pass the actual class? Can we just pass the classname?

Reply:
We need a class since we want to allow developer do some inspection on that class like whether it implements sime interface (Statement, ResultSet, JdbcConnection, etc). Class name wouldn't be enough.